WordPress/gutenberg
# [RNMobile] Buttons block: Display formatting buttons after closing the block settings (only Android)

## What?


This PR fixes an issue on Android related to the Buttons block where the formatting buttons are hidden when opening the block settings and not displayed again after dismissing the settings.

## Why?


Fix https://github.com/WordPress/gutenberg/issues/39491.

## How?


The changes introduced in https://github.com/WordPress/gutenberg/pull/34695 added a workaround for Android to prevent visual glitches when opening the block settings. This is achieved by unfocusing the text input and disabling the button focus when the block settings are opened, which turns to make the formatting buttons to be hidden. This PR updates this logic and checks when any bottom sheet is dismissed to re-enable the button focus and focus the text input again.

## Testing Instructions





**NOTE:** This issue wasn't happening on iOS, so it's recommended to focus the testing on Android.

1. Open a post/page.
1. Add a Buttons block.
1. Observe that the initial Button block is selected.
1. Observe that the formatting buttons in the toolbar are displayed.
1. Tap on the ⚙️ button to open the block settings.
1. Dismiss the bottom sheet.
1. Observe that the formatting buttons in the toolbar are displayed.

## Screenshots or screencast 

iOS|Android
--|--
<video src=https://user-images.githubusercontent.com/14905380/165932252-fc40c07f-7feb-4654-b5a5-a9b8114a710b.mp4>|<video src=https://user-images.githubusercontent.com/14905380/165932442-97e30629-a47a-470d-83de-8256931dfc9e.mp4>
